File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
class Solution(object):
    def topKFrequent(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: List[int]
        """
        ans = []
        freq = {}
        
        for num in nums:
            freq[num] = freq.get(num, 0) +1
        
        for key, v in sorted(freq.items(), key=lambda item: item[1], reverse=True):
            if len(ans) < k:
                ans.append(key)
        return ans

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.cs
public class Solution {
    public int[] TopKFrequent(int[] nums, int k) {
        Dictionary<int, int> freqMap = new Dictionary<int, int>();

        foreach(int num in nums){
            if(!freqMap.ContainsKey(num)){
                freqMap.Add(num, 0);
            }
            ++freqMap[num];
        }

        PriorityQueue<int, int> PQ = new PriorityQueue<int, int>();
        foreach(var kvp in freqMap){
            PQ.Enqueue(kvp.Key, kvp.Value);
            if (PQ.Count > k){
                PQ.Dequeue();
            }
        }

        return PQ.UnorderedItems.Select(el => el.Element).ToArray();
    }
}